lms/djangoapps/teams/static/teams/js/views/teams_tab.js

```diff
@@ -28,27 +60,165 @@
+                                   url: 'teams',
+                                   view: new TempTabView({text: 'This is the new Teams tab.'})
+                               }, {
+                                   title: gettext('Browse'),
+                                   url: 'browse',
+                                   view: new TopicsView({
+                                       collection: this.topicsCollection,
+                                       router: this.router
+                                   })
+                               }],
+                               router: this.router
+                           })
                        });
-                       Backbone.history.start();
                    },
 
                    render: function() {
-                       this.$el.append(this.headerView.$el);
-                       this.headerView.render();
-                       this.$el.append(this.tabbedView.$el);
-                       this.tabbedView.render();
+                       this.mainView.setElement(this.$el).render();
+                       return this;
+                   },
+
+                   /**
+                    * Render the list of teams for the given topic ID.
+                    */
+                   goToTopic: function (topicID) {
+                       // Lazily load the teams-for-topic view in
+                       // order to avoid making an extra AJAX call.
+                       if (this.teamsView && this.teamsView.main.collection.topic_id === topicID) {
+                           this.mainView = this.teamsView;
+                           this.render();
+                       } else {
+                           var topicPromise = this.getTopic(topicID),
+                               self = this;
+                           // The user tried to go to a topic that doesn't exist
+                           topicPromise.done(function (topic) {
+                               var successCb, teamCollection, headerView;
+                               successCb = function (collection) {
+                                   self.mainView = self.teamsView = new ViewWithHeader({
+                                       header: headerView,
+                                       main: new TeamsView({collection: collection, maxTeamSize: self.maxTeamSize})
+                                   });
+                                   self.render();
+                               };
+                               teamCollection = new TeamCollection([], {
+                                   course_id: self.course_id,
+                                   url: self.teams_url,
+                                   topic_id: topicID,
+                                   per_page: 10
+                               });
+                               teamCollection.goTo(1, {
+                                   success: successCb,
```

Comment:
It is a bit strange how this function mixes callbacks and promises. Can you switch everything over to promises?